lib/statsd/instrument/assertions.rb

```diff
@@ -33,39 +37,56 @@ def assert_statsd_key_value(metric_name, options = {}, &block)
 
   private
 
-  def assert_statsd_call(metric_type, metric_name, options = {}, &block)
-    options[:times] ||= 1
+  def assert_statsd_calls(metric_type, metric_name, ordered_metric_options = [], &block)
     metrics = capture_statsd_calls(&block)
     metrics = metrics.select { |m| m.type == metric_type && m.name == metric_name }
     assert metrics.length > 0, "No StatsD calls for metric #{metric_name} were made."
-    assert options[:times] === metrics.length, "The amount of StatsD calls for metric #{metric_name} was unexpected. Expected #{options[:times].inspect}, found #{metrics.length}"
+    expected_calls_count = 0
+    current_metric_index = 0
```

Comment:
You can use `each_with_index` instead of `each` and avoid having this variable declared here like this